x-pack/plugins/monitoring/server/lib/alerts/fetch_ccr_read_exceptions.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { CCRReadExceptionsStats } from '../../../common/types/alerts';
+
+export async function fetchCCRReadExceptions(
+  callCluster: any,
+  index: string,
+  duration: string,
+  size: number
+): Promise<CCRReadExceptionsStats[]> {
+  const params = {
+    index,
+    filterPath: ['aggregations.remote_clusters.buckets'],
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            {
+              nested: {
+                path: 'ccr_stats.read_exceptions',
+                query: {
+                  match_all: {},
+                },
+              },
+            },
+            {
+              term: {
+                type: 'ccr_stats',
+              },
+            },
+            {
+              range: {
+                timestamp: {
+                  gte: `now-${duration}`,
+                },
+              },
+            },
+          ],
+        },
+      },
+      aggs: {
```

Comment:
I don't know if we need to do any aggs actually.

The alert description reads:
> Alert if any CCR read exceptions have been detected

I think we could get away with just searching for the past `{duration}` and only looking at documents that have `read_exceptions` (see my other comment in this file) and return that.

Reply:
We'd just have to make sure we de-dupe the list, which seems like a fairly easy task (using a `Set` or creating a `byId` object and only taking the values)

Reply:
I don't see any benefits of doing it locally. With `aggs` we get "less" data back which I think makes up for its performance degradation (if there are any). Maybe it's fine since we're doing aggregation on filtered data anyways

Reply:
Or, maybe something I'm missing. I'm willing to explore this option, but perhaps as a post/separate task